src/renderer/components/SettingsModal.tsx

```diff
@@ -580,6 +586,49 @@ const PythonSettings = memo(() => {
                                 setOnnxExecutionProvider(String(e.target.value));
                             }}
                         />
+                        {isUsingTensorRt && (
+                            <HStack>
+                                <Toggle
+                                    description="Whether or not to cache the converted TensorRT engines to disk. This can speed up inference times immensely once you have converted once."
+                                    title="Cache TensorRT engines"
+                                    value={onnxShouldTensorRtCache}
+                                    onToggle={() => {
+                                        setOnnxShouldTensorRtCache((prev) => !prev);
+                                    }}
+                                />
+                                <Button
+                                    onClick={() => {
+                                        ipcRenderer
+                                            .invoke('get-appdata')
+                                            .then((appDataPath: string) => {
+                                                const onnxTensorRtCacheLocation =
+                                                    getOnnxTensorRtCacheLocation(appDataPath);
+                                                readdir(onnxTensorRtCacheLocation)
+                                                    .then((files) => {
+                                                        for (const file of files) {
+                                                            unlink(
+                                                                path.join(
+                                                                    onnxTensorRtCacheLocation,
+                                                                    file
+                                                                )
+                                                            ).catch((error) => {
+                                                                log.error(error);
+                                                            });
+                                                        }
+                                                    })
+                                                    .catch((err) => {
+                                                        log.error(err);
+                                                    });
+                                            })
+                                            .catch((err) => {
+                                                log.error(err);
+                                            });
```

Comment:
Let's make this less nested.

```suggestion
                                            .then((appDataPath: string) => {
                                                const onnxTensorRtCacheLocation =
                                                    getOnnxTensorRtCacheLocation(appDataPath);
                                                return readdir(onnxTensorRtCacheLocation)
                                            })
                                            .then((files) => {
                                                for (const file of files) {
                                                    unlink(
                                                        path.join(
                                                            onnxTensorRtCacheLocation,
                                                            file
                                                        )
                                                    ).catch((error) => {
                                                        log.error(error);
                                                    });
                                                }
                                            })
                                            .catch((err) => {
                                                log.error(err);
                                            });
```

Reply:
that doesnt work because the second .then needs to have `onnxTensorRtCacheLocation` available for the path joining

Reply:
Then make it 
```ts
                                            .then(async (appDataPath: string) => {
                                                const onnxTensorRtCacheLocation =
                                                    getOnnxTensorRtCacheLocation(appDataPath);
                                                const files = await readdir(onnxTensorRtCacheLocation);
                                                for (const file of files) {
                                                    unlink(
                                                        path.join(
                                                            onnxTensorRtCacheLocation,
                                                            file
                                                        )
                                                    ).catch((error) => {
                                                        log.error(error);
                                                    });
                                                }
                                            })
                                            .catch((err) => {
                                                log.error(err);
                                            });
```